phoenix-core-server/src/main/java/org/apache/phoenix/index/PhoenixIndexBuilder.java

```diff
@@ -102,7 +102,8 @@ public void batchStarted(MiniBatchOperationInProgress<Mutation> miniBatchOp, Ind
     
     @Override
     public boolean isAtomicOp(Mutation m) {
-        return m.getAttribute(PhoenixIndexBuilderHelper.ATOMIC_OP_ATTRIB) != null;
+        return m.getAttribute(PhoenixIndexBuilderHelper.ATOMIC_OP_ATTRIB) != null ||
```

Comment:
I am slightly confused here. I was under the impression that RETURN_RESULT is only supported for atomic upserts. But this implies non atomic upserts could also get return result.

Reply:
Yes that's correct but this is done to do some preparation to support DELETE in advance. Because with Delete, we don't have ON DUPLICATE KEY clause, nor do we want to support it.

Reply:
If returning results needs to exist independently then `isAtomicOP` API shouldn't change. Rather the caller of the API should check `if (isAtomicOp() || hasReturnResult)`

Reply:
Another reason is let say tomorrow I want to do something just for ON DUPLICATE KEY but not for return result. I cannot use the `isAtomicOp()` API anymore because when it returns true I cannot distinguish between the 2 cases.

Reply:
I was thinking about that and then realized that even if we do it, we sort of are making returnResult as atomic usecase internally (if not for UPSERT, at least for DELETE), that's why i kept it here. WDYT? Shall we still keep it this way or make callers do extra call for where `if (isAtomicOp() || hasReturnResult)` is true?

Reply:
> Another reason is let say tomorrow I want to do something just for ON DUPLICATE KEY but not for return result.

That sounds valid, let me make them different.

Reply:
I think we should keep it separate.